### ISSUE TITLE
Adding a DAISY Sponsorship notice

### ISSUE THREAD

**Issue by NPavie** [daisy/word-save-as-daisy#50]

**Comment by NPavie** [comment]
Summary of the ideas mentioned in the last team meeting :
- Adding the sponsorship message and a "Support our work" button in the ribbon
  - The addin does not have much groups and buttons, so there are plenty of rooms to add such message
  - It would probably be announced last by screen readers, so it should not be really invasive in the user's workflow
  - But it is unsure if the message can be dismissed from the ribbon
- Make the sponsorship message as a system toast notification launched when the addin is loaded, 
  - The message would live in the user notification center for its session time
  - Is usually not displeyd for long so not really intrusive
  - If I am to trust the documentation [here](https://learn.microsoft.com/en-us/windows/apps/design/shell/tiles-and-notifications/send-local-toast?tabs=uwp), we can add a "support our work" button

**Comment by NPavie** [comment]
@bertfrees I tested the use of system notification for the sponsorship message to be displayed at the end of the first conversion launched after the addin is loaded.
So far it works, it's not displayed for long and can be dismissed easily, screen readers announce the notification but it can be a bit messy with the output folder being opened and announced at the same.

The notification is reported on the OS coming from word but i was able to add
- A daisy logo in the message
- An "Attribution text" to mention it's a message from "The DAISY Consortium"
- A sponsorhip request message
- A button that opens the browser (and for now points to the inclusive publishing website that is also used for epubtest)
- I can also make the notification clickable to open an URL directly
  - This could be usefull for when the notification is stored in the notification center (or drawer ? not sure about the term) of Windows : 
  the buttons are hidden by default when notifications are stored there, and clicking on them discard them instead of expanding them by default when they are placed there.

I'm not sure if I chose the right option with the "notify after the first conversion" : 
For users that are closing and reopening word often, it could lead to a lot of notifications from the addin.

I'm thinking instead of notifying the user on launch of the addin directly instead, with a time delay between notifications (like notifying users only once in a week or in a month), and add an option to disable notifications from the addin options form.

This would be a bit less "insistent" ^^'

**Comment by bertfrees** [comment]
Great!

> I can also make the notification clickable to open an URL directly 

I think that could be a good idea. We can have it tested if we're unsure.

> I'm not sure if I chose the right option with the "notify after the first conversion" :
> For users that are closing and reopening word often, it could lead to a lot of notifications from the addin.

But that's what's so great about the notification, it doesn't get in the way, does it? But I can also understand your point. Every time could indeed be a bit much. What about if we were able to detect whether the user clicked the link to go to the website? We could disable the notifications for some time after the user has clicked the link.

Rather than timing how many weeks it is has been since the users last clicked or dismissed the notification, I was thinking perhaps we can use the number of conversions as a measure? It's the power users that work with the software all the time, that we want to urge the most. E.g. dismissing the notification could shut up the notifications for 10 conversions, clicking the link could shut up the notifications for 50 conversions.

> add an option to disable notifications from the addin options form

Yes, it's always nice to have an option like that. I'd like it if I were a user. I think it's important though to not ask the user whether to "not notify him again", but force him to look for the setting. We don't want a user to select "don't notify me again" and then forget about it. We want to make the message clear, by repeating it. If a user is annoyed, he will go look for the option.

**Comment by NPavie** [comment]
> But that's what's so great about the notification, it doesn't get in the way, does it?

Nop, it does not take the hand on the user interaction, it only displasy the message for a few seconds (like maybe 10 or 20) and hides back in the notification center.

> Rather than timing how many weeks it is has been since the users last clicked or dismissed the notification, I was thinking perhaps we can use the number of conversions as a measure?


Hum that's an idea yeah ! I can put that in place with a notification every 10 conversion by default.

I'll have to check if I can do the 50 conversion count on clicking the button or the notification.
From the doc it seems I have limited control on what can be done on both side.
The library provided by Microsoft uses URIs and registered protocols to manage the notifications interactions.

**Comment by NPavie** [comment]
Ok I found a way to do the +50 conversion on the button (binding to a background task that is handled by the addin).

There is only one small doubt I have on the behaviour side : 
If word and the plugin is not opened, it seems to open word in the background to load the addin and run the background task (so there is very noticeable lag before the link is opened, like around 10 seconds)
At first i thought the button did not work, but then it opened the link, and looking at my task manager I saw word running as a background process.

I have the possibility to remove the notification when word is being closed, and I'm thinking it might be best to avoid this "laggy" behavior by dismissing the notifications when word and the addin are closed.
